spec/utils-spec.coffee

```diff
@@ -0,0 +1,83 @@
+Backbone = require 'backbone'
+
+Error = require '../src/error'
+Utils = require '../src/utils'
+
+
+describe 'Brainstem Utils', ->
+  describe '.throwError', ->
+    throwError = ->
+      Utils.throwError('the error')
+
+    context 'Backbone.history.getFragment returns a fragment', ->
+
+      beforeEach ->
+        spyOn(Backbone.history, 'getFragment').and.returnValue('the/fragment#hash')
+
+      it 'throws an error including the message', ->
+        expect(throwError).toThrowError(Error, /the error/)
+
+      it 'throws an error including the fragment', ->
+        expect(throwError).toThrowError(Error, /the\/fragment#hash/)
+
+    context 'Backbone.history.getFragment throws an error', ->
+      beforeEach ->
+        spyOn(Backbone.history, 'getFragment').and.callFake(-> throw new Error('error'))
+
+      it 'throws an error including the message', ->
+        expect(throwError).toThrowError(Error, 'the error')
+
+  describe ".matches", ->
+    it "should recursively compare objects and arrays", ->
+      expect(Utils.matches(2, 2)).toBe true
+      expect(Utils.matches([2], [2])).toBe true, '[2], [2]'
+      expect(Utils.matches([2, 3], [2])).toBe false
+      expect(Utils.matches([2, 3], [2, 3])).toBe true, '[2, 3], [2, 3]'
+      expect(Utils.matches({ hi: "there" }, { hi: "there" })).toBe true, '{ hi: "there" }, { hi: "there" }'
+      expect(Utils.matches([2, { hi: "there" }], [2, { hi: 2 }])).toBe false
+      expect(Utils.matches([2, { hi: "there" }], [2, { hi: "there" }])).toBe true, '[2, { hi: "there" }], [2, { hi: "there" }]'
+      expect(Utils.matches([2, { hi: ["there", 3] }], [2, { hi: ["there", 2] }])).toBe false
+      expect(Utils.matches([2, { hi: ["there", 2] }], [2, { hi: ["there", 2] }])).toBe true, '[2, { hi: ["there", 2] }], [2, { hi: ["there", 2] }]'
+
+  describe ".wrapObjects", ->
+    it "wraps elements in an array with objects unless they are already objects", ->
+      expect(Utils.wrapObjects([])).toEqual []
+      expect(Utils.wrapObjects(['a', 'b'])).toEqual [{a: []}, {b: []}]
+      expect(Utils.wrapObjects(['a', 'b': []])).toEqual [{a: []}, {b: []}]
+      expect(Utils.wrapObjects(['a', 'b': 'c'])).toEqual [{a: []}, {b: [{c: []}]}]
+      expect(Utils.wrapObjects([{'a':[], b: 'c', d: 'e' }])).toEqual [{a: []}, {b: [{c: []}]}, {d: [{e: []}]}]
+      expect(Utils.wrapObjects(['a', { b: 'c', d: 'e' }])).toEqual [{a: []}, {b: [{c: []}]}, {d: [{e: []}]}]
+      expect(Utils.wrapObjects([{'a': []}, {'b': ['c', d: []]}])).toEqual [{a: []}, {b: [{c: []}, {d: []}]}]
```

Comment:
We should consider breaking down these two `it` blocks above further to make it easier to determine issues if/when these tests fail.